master/buildbot/process/builder.py

```diff
@@ -302,12 +302,13 @@ def _startBuildFor(self, slavebuilder, buildrequests):
         cleanups = []
 
         def run_cleanups():
-            try:
-                while cleanups:
-                    fn = cleanups.pop()
-                    fn()
-            except:
-                log.err(failure.Failure(), "while running %r" % (run_cleanups,))
+            while cleanups:
+                fn = cleanups.pop()
+                try:
+                    if fn is not None:
+                        fn()
```

Comment:
Also, why the addition of the "if fn is not None" at all?  All of the items appended to cleanups are lambdas.  I don't see how None could ever be appended.